### PR TITLE
fix: Static analysis issues

### LINKED ISSUE #3204: Static analysis issues
Hello,
We are using fmt as part of spdlog in one of the projects I am working on. Running Klocwork for static analysis against the project's code, several issues are reported in the library. Although many reported cases are false positives, some of them I believe are fixable. I have also checked and most reported issues are related to code that is not available in the newest version of the spdlog (version 1.11.0) and fmt.
Please check the list below.

1. void function returns value
include/spdlog/fmt/bundled/format.h:2448 | operator()()
Code: VOIDRET | Severity: Error (2) | State: Existing | Status: Analyze | Taxonomy: C and C++ | Owner: unowned

Fix:
```c++
...
if (!find<IS_CONSTEXPR>(from, to, Char('}'), p))
{
    handler_.on_text(from, to);
    return ;
}
++p;
...
```

2. Class `specs_setter` defines a copy constructor, but no assignment operator
include/spdlog/fmt/bundled/format.h:1944 | format.h()
Code: CWARN.COPY.NOASSIGN | Severity: Review (4) | State: Existing | Status: Analyze | Taxonomy: C and C++ | Owner: unowned
Fix – removal:
```c++
FMT_CONSTEXPR specs_setter(const specs_setter& other)
    : specs_(other.specs_) {}
```

3. Class `dynamic_specs_handler` defines a copy constructor, but no assignment operator
include/spdlog/fmt/bundled/format.h:2174 | format.h()
Code: CWARN.COPY.NOASSIGN | Severity: Review (4) | State: Existing | Status: Analyze | Taxonomy: C and C++ | Owner: unowned

Fix – removal:
```c++
  FMT_CONSTEXPR dynamic_specs_handler(const dynamic_specs_handler& other)
      : specs_setter<char_type>(other),
        specs_(other.specs_),
        context_(other.context_) {}
```

4. Class `error_handler` defines a copy constructor, but no assignment operator
include/spdlog/fmt/bundled/core.h:407 | core.h()
Code: CWARN.COPY.NOASSIGN | Severity: Review (4) | State: Existing | Status: Analyze | Taxonomy: C and C++ | Owner: unowned

Fix – removal:
```c++
constexpr error_handler(const error_handler&) = default;
```

Please, share your opinion with me.

### PR DESCRIPTION
**Pull request by czudziakm**
fixes #3204 – issues reported by Klocwork

### CODE REVIEW

File: include/fmt/core.h [hunk 1]
```diff
@@ -2718,11 +2711,15 @@ FMT_CONSTEXPR FMT_INLINE void parse_format_string(
       if (from == to) return;
       for (;;) {
         const Char* p = nullptr;
-        if (!find<IS_CONSTEXPR>(from, to, Char('}'), p))
-          return handler_.on_text(from, to);
+        if (!find<IS_CONSTEXPR>(from, to, Char('}'), p)) {
+          handler_.on_text(from, to);
+          return;
```
Comment (vitaut) [inline include/fmt/core.h:2716]: These and changes below look unnecessary, please revert.

Comment (czudziakm) [reply include/fmt/core.h:2716]: I commented on the #3204 reasons for this change and 2 more below.

Comment (vitaut) [reply include/fmt/core.h:2716]: Thanks for providing more context but I still don't see reasons for changing this. Let's revert. The other changes seem fine.

**Comment by vitaut** [comment]
Partly merged in https://github.com/fmtlib/fmt/commit/b0c8263cb26ea178d3a5df1b984e1a61ef578950, thanks.

**Pull request closed without merging**